### ISSUE TITLE
After a tikv network fails, the capacity of a new tikv is expanded, and the leader and region cannot be balanced.

### ISSUE THREAD

**Issue by Lily2025** [tikv/pd#4320]
### What did you do?
1、inject network fault
2、scale out a new tikv（tikv-4）

### What did you expect to see?
the leader and region can be balanced.
NS：endless-oltp--tps-360624-1-365

### What did you see instead?
the leader and region cannot be balanced.
![image](https://user-images.githubusercontent.com/84712107/141233602-55b8e9d3-4e93-4dda-a5b0-0794d88d9589.png)
![image](https://user-images.githubusercontent.com/84712107/141233642-be76f28a-2097-4f2d-a2ff-751ae89879ec.png)

### What version of PD are you using (`pd-server -V`)?
/ # ./pd-server -V
Release Version: v5.3.0
Edition: Community
Git Commit Hash: e8e8c1f4551988ab3fdfd73c13911c998dc14060
Git Branch: heads/refs/tags/v5.3.0
UTC Build Time:  2021-11-09 09:36:28

/ # ./tikv-server -V
TiKV 
Release Version:   5.3.0-alpha
Edition:           Community
Git Commit Hash:   d628f0aa5ff8f6ac258ec9989f6886a1dba34fe4
Git Commit Branch: heads/refs/tags/v5.3.0
UTC Build Time:    2021-11-10 14:30:28
Rust Version:      rustc 1.56.0-nightly (2faabf579 2021-07-27)
Enable Features:   jemalloc mem-profiling portable sse protobuf-codec test-engines-rocksdb cloud-aws cloud-gcp
Profile:           dist_release

**Comment by bufferflies** [comment]
The store4 store heartbeat is wrong: 
 ```
"status": {
        "capacity": "1.433TiB",
        "available": "1.353TiB",
        "used_size": "6.874GiB",
        "leader_count": 206,
        "leader_weight": 1,
        "leader_score": 206,
        "leader_size": 17422,
        "region_count": 206,
        "region_weight": 1,
        "region_score": 17422,
        "region_size": 17422,
        "slow_score": 1,
        "receiving_snap_count": 71,
        "start_ts": "2021-11-10T15:55:13Z",
        "last_heartbeat_ts": "2021-11-11T06:59:18.691694279Z",
        "uptime": "15h4m5.691694279s"
      }
```

the receiving_snap_count greater than 3 , so pd will not select it as target.

**Comment by bufferflies** [comment]
it will fix on https://github.com/tikv/tikv/pull/11342
